sycl/include/sycl/ext/oneapi/experimental/group_load_store.hpp

```diff
@@ -106,6 +108,116 @@ int get_mem_idx(GroupTy g, int vec_or_array_idx) {
     return g.get_local_linear_id() +
            g.get_local_linear_range() * vec_or_array_idx;
 }
+
+// SPIR-V extension:
+// https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/INTEL/SPV_INTEL_subgroups.asciidoc,
+// however it doesn't describe limitations/requirements. Those seem to be
+// listed in the Intel OpenCL extensions for sub-groups:
+// https://registry.khronos.org/OpenCL/extensions/intel/cl_intel_subgroups.html
+// https://registry.khronos.org/OpenCL/extensions/intel/cl_intel_subgroups_char.html
+// https://registry.khronos.org/OpenCL/extensions/intel/cl_intel_subgroups_long.html
+// https://registry.khronos.org/OpenCL/extensions/intel/cl_intel_subgroups_short.html
+// Reads require 4-byte alignment, writes 16-byte alignment. Supported
+// sizes:
+//
+// +--------+-------------+
+// | uchar  | 1,2,4,8,16  |
+// | ushort | 1,2,4,8     |
+// | uint   | 1,2,4,8     |
+// | ulong  | 1,2,4,8     |
+// +--------+-------------+
+//
+// Utility type traits below are used to map user type to one of the block
+// read/write types above.
+
+template <typename IteratorT, std::size_t ElementsPerWorkItem, bool blocked>
+struct BlockInfo {
+  using value_type =
+      remove_decoration_t<typename std::iterator_traits<IteratorT>::value_type>;
+
+  static constexpr int block_size =
+      sizeof(value_type) * (blocked ? ElementsPerWorkItem : 1);
+  static constexpr int num_blocks = blocked ? 1 : ElementsPerWorkItem;
+  static constexpr bool has_builtin =
```

Comment:
A comment explaining the condition required for `has_builtin` would be really helpful. Is there any upper bound on `num_blocks`(or `ElementsPerWorkItem`)?

Reply:
I've added a header to the table and added a comment to `has_builtin`. Upper bound check is already here (discussed offline as well).